amespahdbpythonsuite/mcfitted.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+from typing import Optional
+
+import numpy as np
+import matplotlib.pyplot as plt  # type: ignore
+
+from amespahdbpythonsuite.amespahdb import AmesPAHdb
+
+message = AmesPAHdb.message
+
+
+class MCfitted:
+    """
+    AmesPAHdbPythonSuite mcfitted class.
+    Contains methods to fit and plot the input spectrum using a Monte Carlo approach.
+
+    """
+
+    def __init__(self, d: Optional[dict] = None, **keywords) -> None:
+        self.__set(d, **keywords)
+
+    def set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.set` to parse keywords.
+
+        """
+        self.__set(d, **keywords)
+
+    def __set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Populate data dictionary helper.
+
+        """
+        if isinstance(d, dict):
+            if d.get("type", "") == self.__class__.__name__:
+                if "mcfits" not in keywords:
+                    self.mcfits = d["mcfits"]
+                if "mcpredicted" not in keywords:
+                    self.mcpredicted = d["mcpredicted"]
+                if "distribution" not in keywords:
+                    self.distribuion = d["distribuion"]
+
+        self.mcfits = keywords.get("mcfits", list())
```

Comment:
I just realized that the logic here, and everywhere else we use this paradigm, is wrong ... even if we set self.mcfits above, it is reset to an empty list here ...